### PR TITLE
Make ehrhart_polynomial univariate

### LINKED ISSUE #896: ehrhart Polynomials
Why are they multivariate?
 - easier interface?
 - sparse?
 - they really are sometime multivariate?
The problem/feature is of course that mpoly in 1 variable is completely disjoint from poly...

### PR DESCRIPTION
**Pull request by lkastner**
Is there a constructor I am not seeing that takes two vectors, one for coefficients and one for exponents? Otherwise I will have to write some sanity checks for the output from polymake.

### CODE REVIEW

**Comment by thofma** [comment]
Thanks! And no, there is no such constructor.

**Comment by fieker** [comment]
On Thu, Dec 16, 2021 at 01:26:56AM -0800, Lars Kastner wrote:
> Is there a constructor I am not seeing that takes two vectors, one for coefficients and one for exponents? Otherwise I will have to write some sanity checks for the output from polymake.
No, or not yet? If there is a need we can add

Not good, but

sum(c[i]*x^e[i] for i=1:length(e))

or so should do.
Alternatively, as univariates are always dense

cc = zeros(fmpq, length(c))
for i=1:length(e)
  cc[e[i]] = c[i]
end

then Qx(cc) should work

Different: 
 you're creating PolynomialRing(QQ, "x") thus with cache = true.
 Is this on purpose?
 Shoud we all convene on, e.g. using Hecke.Globals.Qx instead? (for the
 convenient case of just using one ring)

**Comment by lkastner** [comment]
Nevermind, @benlorenz hinted me at a way to get the coefficients padded in the right order.

**Comment by lkastner** [comment]
So last time there was the question how the functions should look like if you want to give them a ring as well. This time I will wait for the conclusion of this discussion.

**Comment by thofma** [comment]
This is perfect, thanks.

**Comment by thofma** [comment]
(But you need to export `h_star_polynomial`, see the doc failure.)

**Comment by lkastner** [comment]
Thanks.

File: src/PolyhedralGeometry/Polyhedron/properties.jl [hunk 1]
```diff
@@ -681,11 +681,51 @@ julia> ehrhart_polynomial(R, c)
...
+
+Compute the $h^*$ polynomial of `P`.
+
+# Examples
+```jldoctest
+julia> c = cube(3)
+A polyhedron in ambient dimension 3
+
+julia> h_star_polynomial(c)
+8*x^3 + 12*x^2 + 6*x + 1
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Polyhedron/properties.jl]:
```suggestion
x^3 + 23*x^2 + 23*x + 1
```

File: src/PolyhedralGeometry/Polyhedron/properties.jl [hunk 2]
```diff
@@ -681,11 +681,51 @@ julia> ehrhart_polynomial(R, c)
...
+# Examples
+```jldoctest
+julia> R, x = PolynomialRing(QQ, "x")
+(Univariate Polynomial Ring in x over Rational Field, x)
+
+julia> c = cube(3)
+A polyhedron in ambient dimension 3
+
+julia> h_star_polynomial(R, c)
+8*x^3 + 12*x^2 + 6*x + 1
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Polyhedron/properties.jl]:
```suggestion
x^3 + 23*x^2 + 23*x + 1
```

**Review by benlorenz — APPROVED** [review]
That example output was supposed to contain the h^* polynomial and not the Ehrhart polynomial.

**Pull request merged**